File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageLoadUrlHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import java.lang.ref.WeakReference;

import jp.hazuki.yuzubrowser.webkit.CustomWebView;

public class WebSrcImageLoadUrlHandler extends WebSrcImageHandler {
    private final WeakReference<CustomWebView> mReference;

    public WebSrcImageLoadUrlHandler(CustomWebView web) {
        mReference = new WeakReference<>(web);
    }

    @Override
    public void handleUrl(String url) {
        CustomWebView web = mReference.get();
        if (web != null)
            web.loadUrl(url);
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageLoadUrlHandler.kt
package jp.hazuki.yuzubrowser.webkit.handler

import jp.hazuki.yuzubrowser.webkit.CustomWebView
import java.lang.ref.WeakReference

class WebSrcImageLoadUrlHandler(web: CustomWebView) : WebSrcImageHandler() {
    private val mReference: WeakReference<CustomWebView> = WeakReference(web)

    override fun handleUrl(url: String) {
        mReference.get()?.loadUrl(url)
    }
}
